### PR TITLE
Refresh rate

### PR DESCRIPTION
**Pull request by imwints**
Closes: #551

### CODE REVIEW

File: src/btop.cpp [hunk 1]
```diff
@@ -108,22 +110,24 @@ namespace Global {
...
-					"usage: btop [-h] [-v] [-/+t] [-p <id>] [--utf-force] [--debug]\n\n"
+					"usage: btop [-h] [-v] [-/+t] [-p <id>] [-r <ms>] [--utf-force] [--debug]\n\n"
 					"optional arguments:\n"
 					"  -h, --help            show this help message and exit\n"
 					"  -v, --version         show version info and exit\n"
 					"  -lc, --low-color      disable truecolor, converts 24-bit colors to 256-color\n"
 					"  -t, --tty_on          force (ON) tty mode, max 16 colors and tty friendly graph symbols\n"
 					"  +t, --tty_off         force (OFF) tty mode\n"
 					"  -p, --preset <id>     start with preset, integer value between 0-9\n"
+					"  -r, --refresh <ms>    set the refresh rate in milliseconds\n"
```
Comment (aristocratos) [inline src/btop.cpp:130]: Would you mind changing `-r` to `-u` and `refresh rate` to `program update rate`, to keep it consistent between how it's named in the config?

**Review by aristocratos — CHANGES_REQUESTED** [review]

File: src/btop.cpp [hunk 2]
```diff
@@ -109,22 +111,24 @@ namespace Global {
...
 
 //* A simple argument parser
-void argumentParser(const int& argc, char **argv) {
+void argumentParser(const int argc, char **argv) {
 	for(int i = 1; i < argc; i++) {
 		const string argument = argv[i];
 		if (is_in(argument, "-h", "--help")) {
 			fmt::println(
-					"usage: btop [-h] [-v] [-/+t] [-p <id>] [--utf-force] [--debug]\n\n"
+					"usage: btop [-h] [-v] [-/+t] [-p <id>] [-r <ms>] [--utf-force] [--debug]\n\n"
```
Comment (aristocratos) [inline src/btop.cpp:123]: `-u <ms>`

File: src/btop.cpp [hunk 3]
```diff
@@ -109,22 +111,24 @@ namespace Global {
...
 					"optional arguments:\n"
 					"  -h, --help            show this help message and exit\n"
 					"  -v, --version         show version info and exit\n"
 					"  -lc, --low-color      disable truecolor, converts 24-bit colors to 256-color\n"
 					"  -t, --tty_on          force (ON) tty mode, max 16 colors and tty friendly graph symbols\n"
 					"  +t, --tty_off         force (OFF) tty mode\n"
 					"  -p, --preset <id>     start with preset, integer value between 0-9\n"
+					"  -u, --refresh <ms>    set the program update rate in milliseconds\n"
```
Comment (aristocratos) [inline src/btop.cpp:131]: Should probably be `-u, --update <ms>` to be more consistent.

File: src/btop.cpp [hunk 4]
```diff
@@ -159,6 +163,19 @@ void argumentParser(const int& argc, char **argv) {
...
+			if (++i >= argc) {
+				fmt::println("ERROR: Refresh option needs an argument");
+				exit(1);
+			}
+			const std::string value = argv[i];
+			if (isint(value)) {
+				Global::arg_refresh = std::clamp(std::stoi(value), 100, 99999);
+			} else {
+				fmt::println("ERROR: Invalid refresh rate");
+				exit(1);
```
Comment (aristocratos) [inline src/btop.cpp:176]: refresh -> update

**Review by aristocratos — CHANGES_REQUESTED** [review]

File: src/btop.cpp [hunk 5]
```diff
@@ -159,6 +163,19 @@ void argumentParser(const int& argc, char **argv) {
...
 			}
 		}
+		else if (is_in(argument, "-u", "--refresh")) {
+			if (++i >= argc) {
+				fmt::println("ERROR: Refresh option needs an argument");
+				exit(1);
+			}
+			const std::string value = argv[i];
+			if (isint(value)) {
+				Global::arg_refresh = std::clamp(std::stoi(value), 100, 99999);
```
Comment (aristocratos) [inline src/btop.cpp:173]: The current max value from config is `86400000`

See https://github.com/aristocratos/btop/blob/f6cab3d3bdc06bec91c5b3681fbb2f00418e17ef/src/btop_config.cpp#L492

Comment (imwints) [reply src/btop.cpp]: Should we give this value a name?

Comment (aristocratos) [reply src/btop.cpp]: Well it's equal to 1 day in ms, so could be written as `1000 * 60 * 60 * 24` to make it more clear.

Comment (imwints) [reply src/btop.cpp]: I think the most readable approach would be to make an named constant like `constexpr int ONE_DAY_MILLIS = 1000 * 60 * 60 * 24;` in Config

Comment (aristocratos) [reply src/btop.cpp]: Sure.

**Review by aristocratos — APPROVED** [review]

**Pull request merged**